### DISCUSSION TITLE
Event when creating new draft in new site

### DISCUSSION THREAD

**Discussion by kristofser** [craftcms/cms#14678] [Q&A] [answered]
Hi,

Is it possible to hook on the event when creating a new entry / draft on another site?

![image](https://github.com/craftcms/cms/assets/5928907/8d7d780e-011f-45f1-834b-78e0dcf00d25)

When we enable a new site (the section's propation method is set to "Let each entry choose which sites it should be saved to"), we want to prefill some fields from an api endpoint, so we only need the newly created entry, but for some reason we get the current entry when triggering the following hook: `EVENT_AFTER_PROPAGATE` or `EVENT_BEFORE_SAVE`

 We tried checking the `$entry->isNew`, `$entry->sender->isNewForSite`,  and `$entry->propagating` but it returns the siteID from current entry and not the newly created one. Likewise for checking if the entry is a draft - we first end up in the entry we were viewing previously - not the one we're creating in the new site.

Looked at this with @janhenckens @carolineboeykens and we couldn't figure it out.

**Comment by brandonkelly** [comment, marked as answer]
When you add a new site to an entry, the entry is saved from the current site, so it’s expected that `siteId` will be the old site from `EVENT_AFTER_PROPAGATE`. But `$entry->newSiteIds` will be set to an array with the newly-added site’s ID, so that’s what you’re going to want to look for.

**Comment by kristofser** [comment]
hey @brandonkelly 
That is exactly what I was looking for.

Thank you for the swift response!
